near/omni-bridge/src/btc.rs

```diff
@@ -0,0 +1,149 @@
+use crate::{ext_token, Contract, ContractExt, Role, FT_TRANSFER_CALL_GAS, ONE_YOCTO};
+use bitcoin::{Address, Network, TxOut};
+use near_plugins::{access_control_any, pause, AccessControllable, Pausable};
+use near_sdk::json_types::U128;
+use near_sdk::{
+    env, near, require, serde_json, AccountId, Gas, Promise, PromiseError, PromiseOrValue,
+};
+use omni_types::btc::TokenReceiverMessage;
+use omni_types::{ChainKind, Fee, TransferId, TransferMessage};
+use std::str::FromStr;
+
+const SUBMIT_TRANSFER_TO_BTC_CONNECTOR_CALLBACK_GAS: Gas = Gas::from_tgas(5);
+
+#[near]
+impl Contract {
+    #[payable]
+    #[pause(except(roles(Role::DAO, Role::UnrestrictedRelayer)))]
+    pub fn submit_transfer_to_utxo_chain_connector(
+        &mut self,
+        transfer_id: TransferId,
+        msg: String,
+        fee_recipient: Option<AccountId>,
+        fee: &Option<Fee>,
+    ) -> Promise {
+        let transfer = self.get_transfer_message_storage(transfer_id);
+
+        let message = serde_json::from_str::<TokenReceiverMessage>(&msg).expect("INVALID MSG");
+        let amount = U128(transfer.message.amount.0 - transfer.message.fee.fee.0);
+
+        if let Some(btc_address) = transfer.message.recipient.get_btc_address() {
+            if let TokenReceiverMessage::Withdraw {
+                target_btc_address,
+                input: _,
+                output,
+            } = message
+            {
+                require!(
+                    btc_address == target_btc_address,
+                    "Incorrect target address"
+                );
+                let output_amount = Self::get_output_amount(&output, &target_btc_address);
```

Comment:
Can we simplify this code by verifying the fee in the btc contract instead?|
@olga24912

Reply:
fixed: https://github.com/Near-One/omni-bridge/pull/348/commits/1bf84ef95e4d9e4e5459b63572b47ce96fd2719a